src/engines/mod.rs

```diff
@@ -383,7 +377,9 @@ impl<'a, I: 'a + IoProvider> ExecutionState<'a, I> {
             }
         }
 
-        panic!("unexpected handle {:?}", handle);
+        tt_warning!(self.status, "input close failed: unexpected handle {:?}", handle);
```

Comment:
Can you add statement making it clearer that this is an internal bug? Something like "serious internal bug: unexpected handle in input close".